### PR TITLE
fix: support square avatar

### PR DESCRIPTION
**Pull request by lelenei**
## Description

Closes # mf-3392

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/mask/src/social-network-adaptor/twitter.com/injection/NFT/NFTAvatarClip.tsx [hunk 1]
```diff
@@ -71,13 +95,77 @@ function NFTAvatarClipInTwitter() {
...
+        const styles = window.getComputedStyle(dom)
+        return styles.clipPath.includes('#shape-square')
+            ? AvatarType.AVATAR_SQUARE
+            : styles.clipPath.includes('#shape-hex')
+            ? AvatarType.AVATAR_CLIP
+            : AvatarType.AVATAR_CIRCLE
+    }, [])
+
     if (isZero(size) || !identity.identifier) return null
+    console.log(avatarType)
```
Comment (guanbinrui) [inline packages/mask/src/social-network-adaptor/twitter.com/injection/NFT/NFTAvatarClip.tsx]: Remove it.

File: packages/mask/src/social-network-adaptor/twitter.com/injection/NFT/NFTAvatarClip.tsx [hunk 2]
```diff
@@ -71,13 +95,77 @@ function NFTAvatarClipInTwitter() {
...
+    const { value = { amount: '0', symbol: 'ETH', name: '', slug: '' }, loading: loadingNFT } = useNFT(
+        account,
+        avatarMetadata?.address,
+        avatarMetadata?.token_id,
+        NetworkPluginID.PLUGIN_EVM,
+        ChainId.Mainnet,
+    )
+
+    const name = useMemo(() => {
+        if (loading || loadingNFT) return 'Loading...'
```
Comment (guanbinrui) [inline packages/mask/src/social-network-adaptor/twitter.com/injection/NFT/NFTAvatarClip.tsx]: i18n

File: packages/mask/src/social-network-adaptor/twitter.com/injection/NFT/NFTAvatarClip.tsx [hunk 3]
```diff
@@ -71,13 +95,77 @@ function NFTAvatarClipInTwitter() {
         }
     }, [location.pathname])
 
+    const avatarType = useMemo(() => {
```
Comment (Jack-Works) [inline packages/mask/src/social-network-adaptor/twitter.com/injection/NFT/NFTAvatarClip.tsx]: this hook use looks suspicious

File: packages/mask/src/social-network-adaptor/twitter.com/injection/NFT/NFTAvatarClip.tsx [hunk 4]
```diff
@@ -72,12 +86,59 @@ function NFTAvatarClipInTwitter() {
...
+
+    const price = useMemo(() => {
+        if (loading || loadingNFT) return 'Loading...'
+        return formatPrice(value.amount, value.symbol)
+    }, [JSON.stringify(value), loading, loadingNFT])
+
+    if (!avatarMetadata?.address || !avatarMetadata?.token_id) return null
+
+    return (
+        <>
```
Comment (guanbinrui) [inline packages/mask/src/social-network-adaptor/twitter.com/injection/NFT/NFTAvatarClip.tsx]: not necessary.

File: packages/mask/src/social-network-adaptor/twitter.com/injection/NFT/NFTAvatarInTwitter.tsx [hunk 5]
```diff
@@ -70,10 +64,61 @@ interface NFTAvatarInTwitterProps {
...
+    useInjectedCSS(showAvatar, updatedAvatar)
+    useUpdatedAvatar(showAvatar, nftAvatar)
+
+    const handlerWatcher = () => {
+        const avatar = searchAvatarSelector().evaluate()?.getAttribute('src')
+        if (!avatar || !nftAvatar?.avatarId) return
+        setUpdatedAvatar(!!nftAvatar?.avatarId && Twitter.getAvatarId(avatar ?? '') === nftAvatar.avatarId)
+    }
+
+    new MutationObserverWatcher(searchAvatarMetaSelector())
```
Comment (guanbinrui) [inline packages/mask/src/social-network-adaptor/twitter.com/injection/NFT/NFTAvatarInTwitter.tsx]: Boom! Memory leak.

File: packages/types/src/types.ts [hunk 6]
```diff
@@ -133,6 +133,8 @@ export namespace SocialNetworkUI {
             postAndReplyNFTAvatar?(signal: AbortSignal): void
             /** @deprecated same reason as useAvatar */
             avatarClipNFT?(signal: AbortSignal): void
+            avatarSquareNFT?(signal: AbortSignal): void
+            avatarCircleNFT?(signal: AbortSignal): void
```
Comment (Jack-Works) [inline packages/types/src/types.ts]: is this necessary? can you merge them into one function?

File: packages/mask/src/social-network-adaptor/twitter.com/injection/NFT/useUpdatedAvatar.ts [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+        }
+
+        const clean = () => {
+            linkParentDom.removeEventListener('click', handler, true)
+        }
+
+        if (!nftAvatar) {
+            clean()
+            return
+        }
```
Comment (Jack-Works) [inline packages/mask/src/social-network-adaptor/twitter.com/injection/NFT/useUpdatedAvatar.ts]:
```suggestion
        if (!nftAvatar) return
```

File: packages/mask/src/social-network-adaptor/twitter.com/injection/NFT/useInjectedCSS.ts [hunk 8]
```diff
@@ -0,0 +1,50 @@
+import { rainbowBorderKeyFrames } from '@masknet/plugin-avatar'
+import { useEffect, useRef } from 'react'
+import { searchTwitterAvatarLinkSelector } from '../../utils/selector.js'
+
+export function useInjectedCSS(showAvatar: boolean, updatedAvatar: boolean) {
+    const rainBowElement = useRef<Element | null>()
```
Comment (Jack-Works) [inline packages/mask/src/social-network-adaptor/twitter.com/injection/NFT/useInjectedCSS.ts]:
```suggestion
    const rainbowElement = useRef<Element | null>()
```

File: packages/types/src/types.ts [hunk 9]
```diff
@@ -132,7 +132,6 @@ export namespace SocialNetworkUI {
             /** @deprecated same reason as userAvatar */
             postAndReplyNFTAvatar?(signal: AbortSignal): void
             /** @deprecated same reason as useAvatar */
```
Comment (guanbinrui) [inline packages/types/src/types.ts]:
```suggestion
```

File: packages/mask/src/social-network-adaptor/twitter.com/injection/NFT/useUpdatedAvatar.ts [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+        if (!showAvatar) return
+
+        const linkParentDom = searchTwitterAvatarLinkSelector().evaluate()?.closest('div')
+        if (!linkParentDom) return
+
+        const handler = (event: MouseEvent) => {
+            if (!nftAvatar?.tokenId || !nftAvatar?.address) return
+            event.stopPropagation()
+            event.preventDefault()
+            if (!nftAvatar.pluginId || !nftAvatar.chainId) return
```
Comment (guanbinrui) [inline packages/mask/src/social-network-adaptor/twitter.com/injection/NFT/useUpdatedAvatar.ts]: ?

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**